examples/ensemble/plot_stack_predictors.py

```diff
@@ -0,0 +1,115 @@
+"""
+=================================
+Combine predictors using stacking
+=================================
+
+Stacking refers to a method to blend estimators. In this strategy, some
+estimators are individually fitted on some training data while a final
+estimator is trained using the stacked predictions of these base estimators.
+
+In this example, we illustrate the use case in which different regressors are
+stacked together and a final linear penalized regressor is used to output the
+prediction. We compare the performance of each individual regressor with the
+stacking strategy. Stacking slightly improves the overall performance.
+
+"""
+print(__doc__)
+
+# Authors: Guillaume Lemaitre <g.lemaitre58@gmail.com>
+# License: BSD 3 clause
+
+###############################################################################
+# The function ``plot_regression_results`` is used to plot the predicted and
+# true targets.
+
+import matplotlib.pyplot as plt
+
+
+def plot_regression_results(ax, y_true, y_pred, title, scores):
+    """Scatter plot of the predicted vs true targets."""
+    ax.plot([y_true.min(), y_true.max()],
+            [y_true.min(), y_true.max()],
+            '--r', linewidth=2)
+    ax.scatter(y_true, y_pred, alpha=0.2)
+
+    ax.spines['top'].set_visible(False)
+    ax.spines['right'].set_visible(False)
+    ax.get_xaxis().tick_bottom()
+    ax.get_yaxis().tick_left()
+    ax.spines['left'].set_position(('outward', 10))
+    ax.spines['bottom'].set_position(('outward', 10))
+    ax.set_xlim([y_true.min(), y_true.max()])
+    ax.set_ylim([y_true.min(), y_true.max()])
+    ax.set_xlabel('Measured')
+    ax.set_ylabel('Predicted')
+    extra = plt.Rectangle((0, 0), 0, 0, fc="w", fill=False,
+                          edgecolor='none', linewidth=0)
+    ax.legend([extra], [scores], loc='upper left')
+    ax.set_title(title)
+
+
+###############################################################################
+# Stack of predictors on a single data set
+###############################################################################
+# It is sometimes tedious to find the model which will best perform on a given
+# dataset. Stacking provide an alternative by combining the outputs of several
+# learners, without the need to choose a model specifically. The performance of
+# stacking is usually close to the best model and sometimes it can outperform
+# the prediction performance of each individual model.
+#
+# Here, we combine 3 learners (linear and non-linear) and use a ridge regressor
+# to combine their outputs together.
+
+from sklearn.ensemble import StackingRegressor
+from sklearn.ensemble import AdaBoostRegressor
+from sklearn.linear_model import LassoCV
+from sklearn.linear_model import RidgeCV
+from sklearn.neighbors import KNeighborsRegressor
+
+estimators = [('KNN', KNeighborsRegressor(n_neighbors=1)),
```

Comment:
Maybe replace KNN with another regressor (e.g., RidgeCV)?
KNN is a bad regressor for this task and maybe we should encourage users to stack "good" estimators?

Reply:
> It is showing that the stack learner is discarding it which I find interesting to show.

Yes, but is it reasonable to throw "bad" estimators into a stacker?
If we decide to keep current version, maybe we can add more comments (e.g., at the beginning of the example)?

Reply:
It depends on what is "bad" estimator. A linear model could perform poorly on some part of data but including it in the stack would be okay since it will be used in the regime where the predictions are right (hopefully).

I have a comment at the end of the example but I am opened to any suggestion.

Reply:
> A linear model could perform poorly on some part of data but including it in the stack would be okay since it will be used in the regime where the predictions are right (hopefully).

If so, maybe the final_estimators should be tree-based model instead of Ridge?

> I have a comment at the end of the example but I am opened to any suggestion.

Sorry I didn't see these comments. I think current example is acceptable.